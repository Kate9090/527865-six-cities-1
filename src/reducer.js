
const initialState = {
  city: `Amsterdam`,
  offerInCity: require(`./mocks/offers-in-amsterdam`).offerHotelList,
};


const ActionCreator = ({
  'NEW_CITY': (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),

  // 'RESET': () => ({
  //   type: `RESET`,
  // }),

});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return Object.assign({}, state, {
        city: action.payload,
        offerInCity: require(`./mocks/offers-in-${state.city}`).offer,
      });
    // case `RESET`:
    //   return Object.assign({}, state, initialState);
  }

  return state;
};

export {reducer, ActionCreator};
