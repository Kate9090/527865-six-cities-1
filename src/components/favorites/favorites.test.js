import React from "react";
import renderer from 'react-test-renderer';

import {BrowserRouter} from 'react-router-dom';
import Favorites from './favorites.jsx';

import {Provider} from 'react-redux';
import reducer from '../../reducer/index';
import {createStore} from 'redux';

const store = createStore(
    reducer
);
it(`renders correctly PlaceCard`, () => {
  const tree = renderer
    .create(<BrowserRouter><Provider store={store}><Favorites
    /></Provider></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
