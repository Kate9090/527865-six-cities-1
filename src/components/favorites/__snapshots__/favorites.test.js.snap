// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly PlaceCard 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <a
              className="header__login"
              href="/login"
              onClick={[Function]}
            >
              Sign in
            </a>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <section
            className="favorites favorites--empty"
          >
            <h1
              className="visually-hidden"
            >
              Favorites (empty)
            </h1>
            <div
              className="favorites__status-wrapper"
            >
              <b
                className="favorites__status"
              >
                Nothing yet saved.
              </b>
              <p
                className="favorites__status-description"
              >
                Save properties to narrow down search or plan yor future trips.
              </p>
            </div>
          </section>
        </ul>
      </section>
    </div>
  </main>,
]
`;
